src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Triggers/EventHubTriggerAttributeBindingProvider.cs

```diff
@@ -59,19 +59,32 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
             string consumerGroup = attribute.ConsumerGroup ?? PartitionReceiver.DefaultConsumerGroupName;
             string resolvedConsumerGroup = _nameResolver.ResolveWholeString(consumerGroup);
 
+            string connectionString = null;
             if (!string.IsNullOrWhiteSpace(attribute.Connection))
             {
                 attribute.Connection = _nameResolver.ResolveWholeString(attribute.Connection);
-                var connectionString = _config.GetConnectionStringOrSetting(attribute.Connection);
+                connectionString = _config.GetConnectionStringOrSetting(attribute.Connection);
                 _options.Value.AddReceiver(resolvedEventHubName, connectionString);
             }
 
             var eventHostListener = _options.Value.GetEventProcessorHost(_config, resolvedEventHubName, resolvedConsumerGroup);
 
+            string storageConnectionString = _config.GetWebJobsConnectionString(ConnectionStringNames.Storage);
+
             Func<ListenerFactoryContext, bool, Task<IListener>> createListener =
              (factoryContext, singleDispatch) =>
              {
-                 IListener listener = new EventHubListener(factoryContext.Executor, eventHostListener, singleDispatch, _options.Value, _logger);
+                 IListener listener = new EventHubListener(
+                                                factoryContext.Descriptor.Id,
+                                                resolvedEventHubName,
+                                                resolvedConsumerGroup,
+                                                connectionString,
+                                                storageConnectionString,
```

Comment:
I made some guesses on the proper way to get these values. I'm not convinced that the event hub connection string would work if the customer provides their own event processor host ([code path here](https://github.com/Azure/azure-functions-eventhubs-extension/blob/dev/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Config/EventHubOptions.cs#L272)) but I couldn't really find any documentation on that feature to test it. I'm not sure where we would get this info in that scenario

Reply:
Right - in Functions scenarios we'll always have connection strings - no EPH provided by the user. However, we need to be sure we handle this situation gracefully if it arises. If we don't have the connection strings for the monitor to do its job, we need to log an error. I think you're already doing that?

Reply:
Yes, it looks like EPH will throw an error, and the connection string operations in the scale monitor constructor will also.